### PR TITLE
Fix 389: document that ShowCreation only works for vectorized mobjects

### LINKED ISSUE #389: Document that ShowCreation does not work on non-VMobjects
It can only work on VMobject. See #387.

Raising NotImplementedError seems like the thing to do. The fix should implement the exception, update the docstring, and change any relevant docs pages.

### PR DESCRIPTION
**Pull request by leotrs**
## List of Changes
- Raise NotImplementedError from ShowPartial when its argument is not a vectorized mobject
- Add some documentation

## Motivation
See #387 and #389. Manim currently throws an error that is not easily understandable I think.

## Further comments
Closes #389

### CODE REVIEW

**Review by huguesdevimeux — APPROVED** [review]
SGTM

**Comment by kilacoda** [comment]
Any reason to use `NotImplementedError` instead of `TypeError`?

**Comment by leotrs** [comment]
Not really. I'm neutral as to which one it should be.

**Review by behackl — APPROVED** [review]
LGTM, although I'd also rather lean towards TypeError (it's not like this is just not implemented, it can't be implemented as we don't have points in this case).

File: manim/animation/creation.py [hunk 1]
```diff
@@ -29,18 +29,51 @@
...
+    --------
+    :class:`ShowCreation`, :class:`~.ShowPassingFlash`
+
     """
-    Abstract class for ShowCreation and ShowPassingFlash
-    """
+
+    def __init__(self, mobject, **kwargs):
+        if not isinstance(mobject, VMobject):
+            raise NotImplementedError(
```
Comment (leotrs) [inline manim/animation/creation.py:47]:
```suggestion
            raise TypeError(
```

File: manim/animation/creation.py [hunk 2]
```diff
@@ -29,18 +29,51 @@
 
 
 class ShowPartial(Animation):
+    """Abstract class for Animations that show the VMobject partially.
+
+    Raises
+    ------
+    :class:`NotImplementedError`
```
Comment (leotrs) [inline manim/animation/creation.py:36]:
```suggestion
    :class:`TypeError`
```

**Comment by leotrs** [comment]
Lol @behackl you beat me to the punch! I was rewriting the squash/merge message :sweat_smile:

**Comment by behackl** [comment]
> Lol @behackl you beat me to the punch! I was rewriting the squash/merge message 😅

Sorry! I thought you were waiting for the pipeline or so. 😅

**Pull request merged**